core/services/bulletprooftxmanager/bulletprooftxmanager.go

```diff
@@ -382,18 +366,18 @@ func SignTx(keyStore KeyStore, address common.Address, tx *gethTypes.Transaction
 		return common.Hash{}, nil, errors.Wrap(err, "SignTx failed")
 	}
 	var hash common.Hash
-	hash, err = signedTxHash(signedTx, chainID)
+	hash, err = signedTxHash(signedTx, chainType)
```

Comment:
What do you think about defining a `chain.Signer()` instead, or a `chain.Sign` function?

That way we can encapsulate some of this messiness.

Reply:
This one seems like a dependency inversion problem again, but packaging up id, config, and keystore together as `ChainKeyStore` (better name?) and converting funcs to methods reduced a lot of repetitive parameter passing.